jwst/wfss_contam/observations.py

```diff
@@ -0,0 +1,545 @@
+                # for JWST images. Will need to figure out wavelength some other way, e.g.
+                # by simply parsing from the filter name.
+                self.fluxes[pivlam] = []
+                dnew = dimage
+                if self.POM_mask01 is not None:
+                    dnew = dimage * self.POM_mask01  # Apply POM transmission mask to the data pixels
+                for i in range(len(self.IDs)):
+                    # This multiplies direct image pixels for each source, which apparently are
+                    # still in units of countrate, by PHOTFLAM, to convert them to fluxes.
+                    # So "fs" is the list of fluxes for all source pixels, at a given wavelength "pivlam"
+                    # Note that JWST data are already flux calibrated, so we've set photflam=1, and
+                    # hence the multiplication does nothing.
+                    self.fluxes[pivlam].append(dnew[self.ys[i], self.xs[i]] * photflam)
+
+            # Use an SED file
+            else:
+                # Need to normalize the object stamps
+                for ID in self.IDs:
+                    vg = self.seg == ID
+                    dnew = dimage
+                    if self.POM_mask01 is not None:
+                        print("Applying POM transmission to data")
+
+                        # Apply POM transmission mask to the data pixels.
+                        # This is a single grey correction for the whole object.
+                        dnew = dimage * self.POM_mask01
+
+                    if self.renormalize is True:
+                        sum_seg = np.sum(dimage[vg])  # But normalize by the whole flux
+                        if sum_seg != 0:
+                            dimage[vg] /= sum_seg
+                    else:
+                        print("not renormlazing sources to unity")
+
+                self.fluxes["SED"] = []
+                for i in range(len(self.IDs)):
+                    self.fluxes["SED"].append(dnew[self.ys[i], self.xs[i]])
+
+    def disperse_all(self, cache=False):
+
+        if cache:
+            print("Object caching ON")
+            self.cache = True
+            self.cached_object = {}
+
+        self.simulated_image = np.zeros(self.dims, np.float)
+
+        for i in range(len(self.IDs)):
+            # if i != 70:
+            #     continue
+
+            if self.cache:
+                self.cached_object[i] = {}
+                self.cached_object[i]['x'] = []
+                self.cached_object[i]['y'] = []
+                self.cached_object[i]['f'] = []
+                self.cached_object[i]['w'] = []
+                self.cached_object[i]['minx'] = []
+                self.cached_object[i]['maxx'] = []
+                self.cached_object[i]['miny'] = []
+                self.cached_object[i]['maxy'] = []
+
+            # Call disperse for this object "i"
+            this_object = self.disperse_chunk(i)
+
+            if self.SBE_save is not None:
```

Comment:
Should this line and everything below be commented out for now as it assumes we are saving to an hdf5 file?

Reply:
Everything related to this SBE file has now been removed. We probably do want to allow for saving an intermediate file like this (in the future), but we'll do so using a standard jwst data model.